src/coreComponents/meshUtilities/CPMesh/CPMeshData.hpp

```diff
@@ -0,0 +1,295 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 Total, S.A
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file CPMeshData.hpp
+ */
+
+
+#ifndef GEOSX_MESHUTILITIES_CPMESH_CPMESHDATA_HPP_
+#define GEOSX_MESHUTILITIES_CPMESH_CPMESHDATA_HPP_
+
+#include "common/DataTypes.hpp"
+#include "dataRepository/ObjectCatalog.hpp"
+
+namespace geosx
+{
+
+namespace CPMesh
+{
+
+/**
+ * @struct CPMeshDimensions
+ * @brief Struct storing global and local mesh dimensions
+ */
+struct CPMeshDimensions
```

Comment:
This class just defines the dimensions of the full mesh and the dimensions of the local MPI partition. It is just a convenience struct storing these dimensions because I need to pass them around all the time to compute local and global indices. But, these dimensions do not correspond to the sub-regions that are later constructed in GEOSX.